auth/README.md

```diff
@@ -0,0 +1,364 @@
+# Authentication
+
+In order to get data from the Vimeo API, we need to generate an access token. There are two types of access tokens - basic and authenticated. It can be obtained by using the [Authenticator](./src/main/java/com/vimeo/networking2/Authenticator.kt) interface. It provides methods to authenticate via email, Google or Facebook. These methods will provide you with an access token via a callback.
+
+## Basic Authentication
+
+A basic access token can be obtained by using your client id, client secret and access scopes. A client id and client secret can created by going to [Create an App](https://developer.vimeo.com/).
+
+After you get your client credentials, create an instance of [ServerConfig](../api/src/main/java/com/vimeo/networking2/config/ServerConfig.kt) as  follows:
+
+```kotlin
+
+// Kotlin
+val serverConfig = ServerConfig(CLIENT_ID, CLIENT_SECRET)
+
+```
+
+```java
+
+// Java
+ServerConfig serverConfig = new ServerConfig(CLIENT_ID, CLIENT_SECRET);
+
+```
+
+In order to authenticate with your client credentials, create an instance of [Authenticator](./src/main/java/com/vimeo/networking2/Authenticator.kt) and pass your `ServerConfig` object to it.
+
+```kotlin
+
+// Kotlin
+val authenticator = Authenticator.create(serverConfig)
+
+```
+
+```java
+
+// Java
+Authenticator authenticator = Authenticator.Factory.create(serverConfig);
+
+```
+
+To authenticate, you would invoke the [Authenticator.clientCredentials](./src/main/java/com/vimeo/networking2/Authenticator.kt#L47) method by supplying a [VimeoCallback](../api/src/main/java/com/vimeo/networking2/VimeoCallback.kt). The callback has a `onSuccess` and `onError` method to provide you the result of the request.
+
+```kotlin
+
+// Kotlin
+authenticator.clientCredentials(object: VimeoCallback<BasicAuthToken>() {
+
+      override fun onSuccess(authResponse: VimeoResponse.Success<BasicAuthToken>) {
+               
+               authResponse.data.accessToken
```

Comment:
This line lacks context, maybe we could do a simple log of the token? Same applies to the errors - although it might make more sense to just put a simple comment like `// handle generic exception`.
```kotlin
println("AccessToken: ${authResponse.data.accessToken}")
```